graphql_compiler/compiler/emit_cypher.py

```diff
@@ -127,10 +148,64 @@ def _emit_with_clause_components(cypher_steps):
     return result
 
 
+def _emit_with_clause_components_for_current_fold_scope(current_fold_scope_cyphersteps):
+    """Emit the component strings for Cypher WITH clause for current fold."""
+    # At this point we should already have emitted all component strings from previous CypherSteps,
+    # whether or not they were in fold scopes or not.
+    result = []
+
+    vertex_names = {}
+    for cypher_step in current_fold_scope_cyphersteps:
+        fold_scope_location = cypher_step.as_block.location
+        full_vertex_name = fold_scope_location_full_path_name(fold_scope_location)
+        vertex_names[u'collect(' + full_vertex_name + ')'] = u'collected_' + full_vertex_name
+
+    # Sort the locations, to ensure a deterministic order.
+    for index, collect_call in enumerate(sorted(vertex_names)):
+        # haha get it? collect_call because we're calling the function collect()?
+        if index > 0:
+            result.append(u',')
+        result.append(u'\n  %(collect_call)s AS %(collected_name)s' %
+                      {'collect_call': collect_call, 'collected_name': vertex_names[collect_call]})
+    return result
+
+
+def _emit_fold_scope(cypher_query):
+    """Return a Cypher query pattern match expression for each fold scope in cypher_query."""
+    query_data = []
+    previous_fold_scope_cypher_steps = []
+    for fold_scope_location in cypher_query.folds:
+        current_fold_scope_cyphersteps = cypher_query.folds[fold_scope_location]
```

Comment:
`cypher_steps` instead of `cyphersteps` for consistency with the rest of the code